sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/NestedDataOperatorConversions.java

```diff
@@ -721,26 +721,7 @@ public DruidExpression toDruidExpression(
 
   public static class JsonObjectOperatorConversion implements SqlOperatorConversion
   {
-    private static final String FUNCTION_NAME = "json_object";
-    private static final SqlFunction SQL_FUNCTION = OperatorConversions
-        .operatorBuilder(FUNCTION_NAME)
-        .operandTypeChecker(OperandTypes.variadic(SqlOperandCountRanges.from(1)))
-        .operandTypeInference((callBinding, returnType, operandTypes) -> {
-          RelDataTypeFactory typeFactory = callBinding.getTypeFactory();
-          for (int i = 0; i < operandTypes.length; i++) {
-            if (i % 2 == 0) {
-              operandTypes[i] = typeFactory.createSqlType(SqlTypeName.VARCHAR);
-              continue;
-            }
-            operandTypes[i] = typeFactory.createTypeWithNullability(
-                typeFactory.createSqlType(SqlTypeName.ANY),
-                true
-            );
-          }
-        })
-        .returnTypeInference(NESTED_RETURN_TYPE_INFERENCE)
```

Comment:
the are some differences between these functions....most importantly the return type is different; the `SqlStdOp` returns `VARCHAR` ; meanwhile this function was returning `NESTED_DATA`

there are also some test failures; one is `CalciteNestedDataQueryTest.testJsonQueryAndJsonObject`

Reply:
@kgyrtkirk i try to use the original OperatorConversions(not SqlStdOperatorTable.JSON_OBJECT), could you help for the following question.
`SELECT JSON_OBJECT(KEY 'n' VALUE JSON_QUERY(nester, '$.n'), KEY 'x' VALUE JSON_VALUE(nest, '$.x')) FROM druid.nested`
this query have not invoke RexToLixTranslator#visitCall, so this one can work as expected.

`select JSON_OBJECT('name': COLUMN_NAME, 'type': \"DATA_TYPE\") from INFORMATION_SCHEMA.COLUMNS order by COLUMN_NAME limit 1`
this query must invoke RexToLixTranslator#visitCall, we have runtime error. do you know what is the reason cause this query must go RexToLixTranslator#visitCall?